### PR TITLE
Feature/observers

### PR DESCRIPTION
**Pull request by rmudambi**

### CODE REVIEW

File: src/vivarium_csu_swissre_cancer/components/observers.py [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+    def on_initialize_simulants(self, pop_data: 'SimulantData'):
+        stratification_groups = pd.Series('', index=pop_data.index)
+
+        self.pipeline_values = {name: pipeline(pop_data.index) for name, pipeline in self.pipelines.items()}
+        self.population_values = self.population_view.get(pop_data.index)
+
+        all_stratifications = self.get_all_stratifications()
+        for stratification in all_stratifications:
+            stratification_group_name = '_'.join([f'{metric["metric"]}_{metric["category"]}'
+                                                  for metric in stratification])
```
Comment (KjellSwedin) [inline src/vivarium_csu_swissre_cancer/components/observers.py:81]: Maybe change "metric" to something like "strat"? There are multiple instances of this construct

`for strat in stratification`

File: src/vivarium_csu_swissre_cancer/components/observers.py [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+        ]
+        self.population_view = builder.population.get_view(columns_required)
+
+        def get_age_range_function(age_lower_bound):
+            return lambda: (
+                (age_lower_bound <= self.population_values['age'])
+                & (self.population_values['age'] < (age_lower_bound + 5))
+            )
+
+        self.stratification_levels = {
```
Comment (collijk) [inline src/vivarium_csu_swissre_cancer/components/observers.py:56]: Age group stratification should be automatically handled by the existing observer tools based on a config switch.

File: src/vivarium_csu_swissre_cancer/components/observers.py [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+    def metrics(self, index: pd.Index, metrics: Dict[str, float]):
+        metrics.update(self.counts)
+        metrics.update(self.person_time)
+        return metrics
+
+    def __repr__(self) -> str:
+        return f"DiseaseObserver({self.disease})"
+
+
+def get_state_person_time(pop: pd.DataFrame, config: Dict[str, bool],
```
Comment (collijk) [inline src/vivarium_csu_swissre_cancer/components/observers.py:316]: I think we want to look at pushing these back into vph.  We've been trading in these functions for a while.

File: src/vivarium_csu_swissre_cancer/components/observers.py (hunk 2 above)
Comment (rmudambi) [reply src/vivarium_csu_swissre_cancer/components/observers.py:56]: These are age cohorts based on entrance age, so we need to stratify this separately

**Pull request merged**